### PR TITLE
Build a backend plugin data source updates

### PR DESCRIPTION
**Pull request by marefr**
Hopefully improves the tutorial a bit, but is still a bit weird to follow. We can work more on this next week. 

I've verified that with the following instructions you will end up with a backend data source plugin that works with Grafana Alerting.

https://github.com/grafana/simple-datasource-backend have been updated, but can still use some additional love in regards to the actual code/functionality.

### CODE REVIEW

**Comment by kylebrandt** [comment]
Should we have a allow_unsigned grafana conf step in this (until we get signing of community backend plugins)?

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 1]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+go get -u github.com/grafana/grafana-plugin-sdk-go
+```
+
+Build backend plugin binaries for Linux, Windows and Darwin to _dist_ directory:
+
+```bash
+mage -v
 ```
 
+Now, lets verify that the plugin you've built so far can be used in Grafana when creating a new data source:
```
Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:77]:
```suggestion
Now, let's verify that the plugin you've built so far can be used in Grafana when creating a new data source:
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 2]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+
+```bash
+mage -v
 ```
 
+Now, let's verify that the plugin you've built so far can be used in Grafana when creating a new data source:
+
+1. Restart your Grafana instance.
+1. Open Grafana in your web browser.
+1. Navigate via the sidemenu to Configuration -> Data Sources.
```
Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:81]:
```suggestion
1. Navigate via the side-menu to Configuration -> Data Sources.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 3]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+1. Click on _Add data source_ button.
```
Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:82]:
```suggestion
1. Click on the _Add data source_ button.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 4]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+1. Find your newly created plugin and select it.
+1. Enter a suitable name and then click the _Save & Test_ button (ignores any errors reported for now).
+
+You should now have created a data source instance of your plugin and should be ready to use it in a dashboard:
+
+1. Navigate via the sidemenu to Create -> Dashboard.
```
Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:88]:
```suggestion
1. Navigate via the side-menu to Create -> Dashboard.
```

**Comment by marefr** [comment]
@kylebrandt good point. If following `Set up Environment and Docker` it will not work due to signing required. 

I think we definitely need to document it in https://grafana.com/docs/grafana/latest/developers/plugins/backend/.

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 5]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+Now, let's verify that the plugin you've built so far can be used in Grafana when creating a new data source:
+
+1. Restart your Grafana instance.
+1. Open Grafana in your web browser.
+1. Navigate via the sidemenu to Configuration -> Data Sources.
+1. Click on _Add data source_ button.
+1. Find your newly created plugin and select it.
+1. Enter a suitable name and then click the _Save & Test_ button (ignores any errors reported for now).
+
+You should now have created a data source instance of your plugin and should be ready to use it in a dashboard:
```
Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:86]:
```suggestion
At this point you should have a newly created data source instance of your plugin, and it should be ready to use it in a dashboard:
```

Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:86]:
```suggestion
At this point you should have a newly created data source instance of your plugin, and it should be ready to use in a dashboard:
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 6]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+
+1. Navigate via the sidemenu to Create -> Dashboard.
+1. Click on _Add new panel_.
+1. In the query tab, select the data source you created.
```
Comment (jessabe) [inline content/tutorials/build-a-data-source-backend-plugin.md:90]:
```suggestion
1. In the query tab, select the data source you just created.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 7]
```diff
@@ -78,16 +78,16 @@ Now, let's verify that the plugin you've built so far can be used in Grafana whe
...
 1. Open Grafana in your web browser.
-1. Navigate via the sidemenu to Configuration -> Data Sources.
-1. Click on _Add data source_ button.
+1. Navigate via the side-menu to Configuration -> Data Sources.
+1. Click on the _Add data source_ button.
 1. Find your newly created plugin and select it.
 1. Enter a suitable name and then click the _Save & Test_ button (ignores any errors reported for now).
 
-You should now have created a data source instance of your plugin and should be ready to use it in a dashboard:
+At this point you should have a newly created data source instance of your plugin, and it should be ready to use in a dashboard:
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:86]:
```suggestion
You now have a new data source instance of your plugin that is ready to use in a dashboard:
```

Avoid using "should" in documentation.

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 8]
```diff
@@ -78,16 +78,16 @@ Now, let's verify that the plugin you've built so far can be used in Grafana whe
 
 1. Restart your Grafana instance.
 1. Open Grafana in your web browser.
-1. Navigate via the sidemenu to Configuration -> Data Sources.
-1. Click on _Add data source_ button.
+1. Navigate via the side-menu to Configuration -> Data Sources.
+1. Click on the _Add data source_ button.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:82]:
```suggestion
1. Click **Add data source**.
```

Use bold when referencing UI elements.

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 9]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
 
 The easiest way to get started is to clone one of our test data datasources. Navigate to the plugin folder that you configured in step 1 and type:
 
 ```
-npx @grafana/toolkit plugin:create my-plugin
+npx @grafana/toolkit@next plugin:create my-plugin
 ```
 
-Select `Backend Datasource Plugin` and follow the rest of the steps in the plugin scaffolding command.
+Select _Backend Datasource Plugin_ and follow the rest of the steps in the plugin scaffolding command.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:52]:
```suggestion
Select **Backend Datasource Plugin** and follow the rest of the steps in the plugin scaffolding command.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 10]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+
+```bash
+mage -v
 ```
 
+Now, let's verify that the plugin you've built so far can be used in Grafana when creating a new data source:
+
+1. Restart your Grafana instance.
+1. Open Grafana in your web browser.
+1. Navigate via the side-menu to Configuration -> Data Sources.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:81]:
```suggestion
1. Navigate via the side-menu to **Configuration** -> **Data Sources**.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 11]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+1. Click on the _Add data source_ button.
+1. Find your newly created plugin and select it.
+1. Enter a suitable name and then click the _Save & Test_ button (ignores any errors reported for now).
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:84]:
```suggestion
1. Enter a name and then click **Save & Test** (ignore any errors reported for now).
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 12]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+
+At this point you should have a newly created data source instance of your plugin, and it should be ready to use in a dashboard:
+
+1. Navigate via the side-menu to Create -> Dashboard.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:88]:
```suggestion
1. Navigate via the side-menu to **Create** -> **Dashboard**.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 13]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+1. Click on _Add new panel_.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:89]:
```suggestion
1. Click **Add new panel**.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 14]
```diff
@@ -35,61 +41,90 @@ In this tutorial, you'll:
...
+1. In the query tab, select the data source you just created.
+1. You should now see a line graph rendered with one series consisting of two data points.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:91]:
```suggestion
1. A line graph is rendered with one series consisting of two data points.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 15]
```diff
@@ -100,12 +135,41 @@ You can read more about how to [build data frames in our docs](https://grafana.c
 {{< tutorials/step title="Add support for health checks" >}}
 
 Implementing the health check handler allows Grafana to verify that a data source has been configured correctly.
-After a user has created a new datasource in Grafana's UI, she can click the **Test** to verify that it works as expected.
+When editing a data source in Grafana's UI, you can click the _Save & Test_ button to verify that it works as expected.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:138]:
```suggestion
When editing a data source in Grafana's UI, you can **Save & Test** to verify that it works as expected.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 16]
```diff
@@ -100,12 +135,41 @@ You can read more about how to [build data frames in our docs](https://grafana.c
...
 
 In this sample data source, there is a 50% chance that the health check will be successful. Make sure to return appropriate error messages to
 the users, informing them about what is misconfigured in the data source.
 
-Open `/pkg/datasource/sample-datasource.go` and navigate to the `CheckHealth` function to see how the health check for this sample plugin is implemented.
+Open `/pkg/sample-plugin.go`. In this file you will see the `SampleDatasource` struct also implements the [backend.CheckHealthHandler](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend?tab=doc#CheckHealthHandler) interface. Navigate to the `CheckHealth` method to see how the health check for this sample plugin is implemented.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:143]:
```suggestion
Open `/pkg/sample-plugin.go`. In this file you'll see that the `SampleDatasource` struct also implements the [backend.CheckHealthHandler](https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/backend?tab=doc#CheckHealthHandler) interface. Navigate to the `CheckHealth` method to see how the health check for this sample plugin is implemented.
```

File: content/tutorials/build-a-data-source-backend-plugin.md [hunk 17]
```diff
@@ -100,12 +135,41 @@ You can read more about how to [build data frames in our docs](https://grafana.c
...
+
+{{< /tutorials/step >}}
+{{< tutorials/step title="Enable Grafana Alerting" >}}
+
+1. Open _src/plugin.json_.
+1. Add the top level _backend_ property with a value of _true_ to specify that your plugin supports Grafana Alerting, e.g.
```
Comment (marcusolsson) [inline content/tutorials/build-a-data-source-backend-plugin.md:149]:
```suggestion
1. Add the top level `backend` property with a value of `true` to specify that your plugin supports Grafana Alerting, e.g.
```

**Pull request merged**